server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -387,9 +401,15 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
         httpClientStatsTracker.updateClientStats(httpRequest, httpChannel);
+        final RequestTrackingHttpChannel trackingChannel = httpChannels.get(httpChannel);
         final long startTime = threadPool.rawRelativeTimeInMillis();
         try {
-            handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+            if (trackingChannel != null) {
+                // The channel may not be present if the close listener (set in serverAcceptedChannel) runs before this method because the
+                // connection closed early
+                trackingChannel.incomingRequest();
+            }
```

Comment:
Can we log at debug level this edge case where the channel was closed just as a new incoming request was accepted?

Reply:
Added.